specification/apicenter/data-plane/ApiCenter.DataApi/preview/2024-02-01-preview/azure.apicenter.dataapi.json

```diff
@@ -0,0 +1,1644 @@
+      }
+    },
+    "PagedApi": {
+      "type": "object",
+      "description": "Paged collection of Api items",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The Api items on this page",
+          "items": {
+            "$ref": "#/definitions/Models.Api"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "PagedApiDefinition": {
+      "type": "object",
+      "description": "Paged collection of ApiDefinition items",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The ApiDefinition items on this page",
+          "items": {
+            "$ref": "#/definitions/Models.ApiDefinition"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "PagedApiDeployment": {
+      "type": "object",
+      "description": "Paged collection of ApiDeployment items",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The ApiDeployment items on this page",
+          "items": {
+            "$ref": "#/definitions/Models.ApiDeployment"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "PagedApiVersion": {
+      "type": "object",
+      "description": "Paged collection of ApiVersion items",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The ApiVersion items on this page",
+          "items": {
+            "$ref": "#/definitions/Models.ApiVersion"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "PagedEnvironment": {
+      "type": "object",
+      "description": "Paged collection of Environment items",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The Environment items on this page",
+          "items": {
+            "$ref": "#/definitions/Models.Environment"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
```

Comment:
Since collections received as GET operation, `required` doesn't make sense here:
```
      "required": [
        "value"
      ]
```

Reply:
This response contract is automatically generated by the `Operations.ResourceList` Azure Core standard resource operation and as Mike mentioned in another comment, "there is no problem with required properties in responses and actually they are quite helpful as they indicate which properties will always be present in the response. ARM previously disallowed this but recently changed to permit it." So I think it is OK to keep it here as it's auto-generated.

Reply:
Actually guideline recommend this. It means user can always take e.g. `responseBody.value.size` even if there is not result.